clarity/contracts/arkadiko-liquidation-rewards-v1-2.clar

```diff
@@ -0,0 +1,232 @@
+  (let (
+    (reward-info (get-reward-data reward-id))
+    (share-block (get share-block reward-info))
+    (total-amount (get total-amount reward-info))
+
+    (user-shares (unwrap-panic (contract-call? liquidation-pool get-shares-at user share-block)))
+    (total-rewards (/ (* user-shares total-amount) u10000000))
+
+    (claimed (get claimed (get-reward-claimed reward-id user)))
+  )
+    (asserts! (is-eq (contract-of liquidation-pool) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "liquidation-pool"))) (ok u0))
+
+    (if claimed 
+      (ok u0)
+      (ok total-rewards)
+    )
+  )
+)
+
+;; @desc claim rewards for user
+;; @param reward-id; the rewards
+;; @param token; the reward token
+;; @param liquidation-pool; the pool on which shares are based
+(define-public (claim-rewards-of (reward-id uint) (token <ft-trait>) (liquidation-pool <liquidation-pool-trait>))
+  (let (
+    (user tx-sender)
+    (reward-amount (unwrap-panic (get-rewards-of user reward-id liquidation-pool)))
+    (reward-info (get-reward-data reward-id))
+  )
+    (asserts! (not (get-shutdown-activated)) (err ERR-EMERGENCY-SHUTDOWN-ACTIVATED))
+    (asserts! (is-eq (contract-of token) (get token reward-info)) (err ERR-WRONG-TOKEN))
+    (asserts! (> burn-block-height (get unlock-block reward-info)) (err ERR-REWARD-LOCKED))
+
+    (if (is-eq reward-amount u0)
+      (ok u0)
+      (begin 
+        (if (get token-is-stx reward-info)
+          (try! (as-contract (stx-transfer? reward-amount (as-contract tx-sender) user)))
+          (try! (as-contract (contract-call? token transfer reward-amount (as-contract tx-sender) user none)))
+        )
+
+        (map-set reward-claimed  { reward-id: reward-id, user: user } { claimed: true })
+
+        (ok reward-amount)
+      )
+    )
+  )
+)
+
+;; ---------------------------------------------------------
+;; Admin
+;; ---------------------------------------------------------
+
+(define-public (update-reward-tokens (token principal) (whitelisted bool))
+  (begin
+    (asserts! (is-eq tx-sender (contract-call? .arkadiko-dao get-dao-owner)) (err ERR-NOT-AUTHORIZED))
+
+    (map-set reward-tokens  { token: token } { whitelisted: whitelisted })
+    
+    (ok true)
+  )
+)
+
+(define-public (toggle-shutdown)
+  (begin
+    (asserts! (is-eq tx-sender (contract-call? .arkadiko-dao get-guardian-address)) (err ERR-NOT-AUTHORIZED))
+
+    (ok (var-set shutdown-activated (not (var-get shutdown-activated))))
+  )
+)
+
+;; ---------------------------------------------------------
+;; Init
+;; ---------------------------------------------------------
+
+(begin
+  (map-set reward-tokens  { token: .arkadiko-token } { whitelisted: true })
+  (map-set reward-tokens  { token: .xstx-token } { whitelisted: true })
+
+  ;; TODO - UPDATE ADDRESS FOR MAINNET
+  (map-set reward-tokens  { token: .Wrapped-Bitcoin } { whitelisted: true })
+)
```

Comment:
Q: add atALEX?

Reply:
Q: Yes, it's done in the deployed contract:
https://explorer.stacks.co/txid/0xd4f083a5708bc6a70fbfbc8bd6b2f647f36f72d4714a194b8a1eb600d1e7c665?chain=mainnet